ui/app/store/actions.js

```diff
@@ -1178,15 +1178,15 @@ export function setSelectedToken (tokenAddress) {
 }
 
 export function setSelectedAddress (address) {
-  return (dispatch) => {
+  return async (dispatch) => {
     dispatch(showLoadingIndication())
     log.debug(`background.setSelectedAddress`)
-    background.setSelectedAddress(address, (err) => {
+    try {
+      await promisifiedBackground.setSelectedAddress(address)
+    } catch (error) {
       dispatch(hideLoadingIndication())
-      if (err) {
-        return dispatch(displayWarning(err.message))
-      }
-    })
+      return dispatch(displayWarning(error.message))
```

Comment:
Should we separate this into two lines—I believe the old return value was `undefined`, even when there was an error

Reply:
Good catch! I meant to check for this since you pointed this out in the other PR.... I forgot with these two.

I'll just remove the `return`, since there's no `return` elsewhere either, and nothing follows this block.

Reply:
This has been updated